### PR TITLE
A5-complete-review

### PR DESCRIPTION
**Pull request by swatikadivar**

### CODE REVIEW

File: server/aesdsocket.c [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+
+void cleanup() 
+{
+  if (server_addr != NULL) {
+    freeaddrinfo(server_addr);
+  }
+
+  // check if file exists and if so, delete it
+  if (access(TEMPFILE, F_OK) == 0)  {
+    remove(TEMPFILE);
```
Comment (swatikadivar) [inline server/aesdsocket.c:389]: @jmichael16  Is this function safe to do in signal handler?

File: server/aesdsocket.c [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+      if (ret == -1 && errno != EINTR) {
+        LOG(LOG_ERR, "recv returned -1"); perror("recv()");
+        free(recv_buf); recv_buf_size = 0; recv_buf = NULL;
+        return -1;
+      } else if (ret == 0) { // end of file (peer socket shutdown)
+        LOG(LOG_INFO, "Peer socket shutdown");
+        state = ACCEPTING_CONNECTIONS; 
+        break;
+      } else if (ret > 0) {
+        recv_buf = realloc(recv_buf, recv_buf_size + ret);
```
Comment (swatikadivar) [inline server/aesdsocket.c:290]: @jmichael16 what assumption is made here about the size of RAM heap relative to the total size of all packets sent?

**Review by swatikadivar — COMMENTED** [review]
Nice work!

File: server/aesdsocket.c (hunk 2 above)
Comment (jmiv-eng) [reply server/aesdsocket.c:290]: The heap only gets as big as the single packet of interest. You can see it is reallocd back to zero after writing to the file.

File: server/aesdsocket.c (hunk 1 above)
Comment (jmiv-eng) [reply server/aesdsocket.c:389]: No, you are correct, remove() is not signal safe. I have fixed in future assignment when moving to the threading implementation.

File: server/aesdsocket.c (hunk 2 above)
Comment (swatikadivar) [reply server/aesdsocket.c:290]: It is working in this case but what if you're getting a packet size > 1GB and your heap is of size 800MB?
@dwalkes Can you please review this code? he is doing realloc till one packet size but may fail if packet size is more than heap allocated. Am i right?

Comment (jmiv-eng) [reply server/aesdsocket.c:290]: According to the A5 instructions (Part e): "You may assume the length of the packet will be shorter than the available heap size."

Comment (swatikadivar) [reply server/aesdsocket.c:290]: okay but what if realloc fails? 🤔 error check is missing

**Pull request closed without merging**